src/interactivity/base/convert.cpp

```diff
@@ -0,0 +1,210 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+#include "precomp.h"
+#include "../inc/convert.hpp"
+#include "../../types/inc/convert.hpp"
+
+#ifdef BUILD_ONECORE_INTERACTIVITY
+#include "../inc/VtApiRedirection.hpp"
+#endif
+
+#pragma hdrstop
+
+// TODO: MSFT 14150722 - can these const values be generated at
+// runtime without breaking compatibility?
+static constexpr WORD altScanCode = 0x38;
+static constexpr WORD leftShiftScanCode = 0x2A;
+
+std::deque<std::unique_ptr<KeyEvent>> Microsoft::Console::Interactivity::CharToKeyEvents(const wchar_t wch,
+                                                                                         const unsigned int codepage)
+{
+    const short invalidKey = -1;
+    short keyState = VkKeyScanW(wch);
+
+    if (keyState == invalidKey)
+    {
+        // Determine DBCS character because these character does not know by VkKeyScan.
+        // GetStringTypeW(CT_CTYPE3) & C3_ALPHA can determine all linguistic characters. However, this is
+        // not include symbolic character for DBCS.
+        WORD CharType = 0;
+        GetStringTypeW(CT_CTYPE3, &wch, 1, &CharType);
+
+        if (WI_IsFlagSet(CharType, C3_ALPHA) || GetQuickCharWidth(wch) == CodepointWidth::Wide)

```

Comment:
this is highly suspect -- the width of a character should not impact its encoding in key events. HOWEVER: this is NOT NEW CODE and I WILL NOT FIX IT HERE.

Reply:
Yeah, don't fix it. But you're right. The width... shouldn't... ughhhhhh.